pkg/streamingpromql/types/hpoint_ring_buffer.go

```diff
@@ -89,8 +89,19 @@ func (b *HPointRingBuffer) CopyPoints(maxT int64) ([]promql.HPoint, error) {
 		return nil, err
 	}
 
-	combined = append(combined, head...)
-	combined = append(combined, tail...)
+	combine := func(p []promql.HPoint) {
```

Comment:
Might be good to expand the docstring for `CopyPoints` to explain that it duplicates the `FloatHistogram` values as well.